File: Python with BDD Cucumber/tests/pages/dashboard_page.py
from selenium.webdriver.common.by import By
from tests.utils.base_page import BasePage
from tests.utils.base_element import BaseElement


class DashboardPage(BasePage):
    """Dashboard page objects for ParaBank"""
    
    # Locators
    WELCOME_MESSAGE = (By.CSS_SELECTOR, "h2")
    LOGOUT_LINK = (By.LINK_TEXT, "Log Out")
    ACCOUNTS_OVERVIEW_LINK = (By.LINK_TEXT, "Accounts Overview")
    TRANSFER_FUNDS_LINK = (By.LINK_TEXT, "Transfer Funds")
    BILL_PAY_LINK = (By.LINK_TEXT, "Bill Pay")
    REQUEST_LOAN_LINK = (By.LINK_TEXT, "Request Loan")
    UPDATE_PROFILE_LINK = (By.LINK_TEXT, "Update Contact Info")
    
    def __init__(self, driver):
        super().__init__(driver)
    
    def is_dashboard_displayed(self):
        """Check if dashboard is displayed"""
        element = BaseElement(self.driver, self.WELCOME_MESSAGE)
        return element.is_displayed()
    
    def get_welcome_message(self):
        """Get welcome message"""
        element = BaseElement(self.driver, self.WELCOME_MESSAGE)
        return element.get_text()
    
    def click_logout(self):
        """Click logout link"""
        element = BaseElement(self.driver, self.LOGOUT_LINK)
        element.click()
    
    def click_accounts_overview(self):
        """Click accounts overview link"""
        element = BaseElement(self.driver, self.ACCOUNTS_OVERVIEW_LINK)
        element.click()
    
    def click_transfer_funds(self):
        """Click transfer funds link"""
        element = BaseElement(self.driver, self.TRANSFER_FUNDS_LINK)
        element.click()
    
    def click_bill_pay(self):
        """Click bill pay link"""
        element = BaseElement(self.driver, self.BILL_PAY_LINK)
        element.click()
    
    def click_request_loan(self):
        """Click request loan link"""
        element = BaseElement(self.driver, self.REQUEST_LOAN_LINK)
        element.click()
    
    def click_update_profile(self):
        """Click update profile link"""
        element = BaseElement(self.driver, self.UPDATE_PROFILE_LINK)
        element.click()

File: Java with BDD Cucumber/src/main/java/com/parabank/pages/DashboardPage.java
package com.parabank.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import com.parabank.base.BaseClass;

/**
 * Dashboard/Accounts Page Object Model
 */
public class DashboardPage extends BaseClass {
    // Locators
    private By accountsOverviewLink = By.xpath("//a[contains(text(), 'Accounts Overview')]");
    private By openNewAccountLink = By.xpath("//a[contains(text(), 'Open New Account')]");
    private By transferFundsLink = By.xpath("//a[contains(text(), 'Transfer Funds')]");
    private By accountsList = By.xpath("//table[@class='accounts']/tbody/tr");
    private By logoutLink = By.xpath("//a[contains(text(), 'Log Out')]");
    private By welcomeText = By.cssSelector("h2");
    private By accountTypeDropdown = By.id("type");
    private By fromAccountDropdown = By.id("fromAccountId");
    private By toAccountDropdown = By.id("toAccountId");
    private By transferAmountField = By.id("amount");
    private By transferButton = By.xpath("//input[@value='Transfer']");
    private By createAccountButton = By.xpath("//input[@value='Open New Account']");
    private By confirmationMessage = By.cssSelector("div#rightPanel p");

    public DashboardPage(WebDriver driver) {
        this.setDriver(driver);
    }

    /**
     * Navigate to accounts overview
     */
    public void navigateToAccountsOverview() {
        clickElement(accountsOverviewLink);
    }

    /**
     * Navigate to open new account page
     */
    public void navigateToOpenNewAccount() {
        clickElement(openNewAccountLink);
    }

    /**
     * Navigate to transfer funds page
     */
    public void navigateToTransferFunds() {
        clickElement(transferFundsLink);
    }

    /**
     * Select account type
     */
    public void selectAccountType(String type) {
        selectByVisibleText(accountTypeDropdown, type);
    }

    /**
     * Select from account
     */
    public void selectFromAccount(String accountId) {
        selectByValue(fromAccountDropdown, accountId);
    }

    /**
     * Select to account
     */
    public void selectToAccount(String accountId) {
        selectByValue(toAccountDropdown, accountId);
    }

    /**
     * Enter transfer amount
     */
    public void enterTransferAmount(String amount) {
        typeText(transferAmountField, amount);
    }

    /**
     * Click transfer button
     */
    public void clickTransferButton() {
        clickElement(transferButton);
    }

    /**
     * Click create account button
     */
    public void clickCreateAccountButton() {
        clickElement(createAccountButton);
    }

    /**
     * Get confirmation message
     */
    public String getConfirmationMessage() {
        return getText(confirmationMessage);
    }

    /**
     * Get welcome message
     */
    public String getWelcomeMessage() {
        return getText(welcomeText);
    }

    /**
     * Check if accounts list is displayed
     */
    public boolean isAccountsListDisplayed() {
        return isElementDisplayed(accountsList);
    }

    /**
     * Logout
     */
    public void logout() {
        clickElement(logoutLink);
    }
}
